src/content/components/Tooltip.js

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import ReactDOM from 'react-dom';
+import type { CssPixels } from '../../common/types/units';
+
+require('./Tooltip.css');
+
+const MOUSE_OFFSET = 15;
+
+type Props = {
+  mouseX: CssPixels,
+  mouseY: CssPixels,
+  children?: React$Element<*>,
+}
+
+export default class Tooltip extends PureComponent {
+  props: Props;
+
+  state: {
+    interiorElement: HTMLElement | null,
+    interiorElementRAF: HTMLElement | null,
+  };
+
+  _isMounted: boolean;
+  _mountElement: ?HTMLElement;
+
+  constructor(props: Props) {
+    super(props);
+    (this: any)._getMountElement = this._getMountElement.bind(this);
+    this.state = {
+      interiorElement: null,
+      interiorElementRAF: null,
+    };
+  }
+
+  _getMountElement(el: HTMLElement) {
+    this.setState({ interiorElement: el });
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    // Create a DOM node outside of the normal heirarchy.
+    const el = document.createElement('div');
+    el.className = 'tooltipMount';
+
+    // Satisfy flow null checks.
+    if (!document.body) {
+      throw new Error('No document body was found to append to.');
+    }
+    document.body.appendChild(el);
+    this._mountElement = el;
+    this._renderTooltipContents();
+  }
+
+  componentWillUnmount() {
+    ReactDOM.unmountComponentAtNode(this._mountElement);
+    // Satisfy flow null checks.
+    if (this._mountElement) {
+      this._mountElement.remove();
+    }
+    this._isMounted = false;
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.children !== this.props.children) {
+      this.setState({interiorElementRAF: null });
+      this._allowInteriorElementLayout();
+    }
+  }
+
+  componentDidUpdate() {
+    this._allowInteriorElementLayout();
+    this._renderTooltipContents();
+  }
+
+  _allowInteriorElementLayout() {
+    const { interiorElement, interiorElementRAF } = this.state;
+    if (interiorElement && !interiorElementRAF) {
+      // Allow the interior element to fully lay out, then update the sizing.
+      requestAnimationFrame(() => {
+        if (this._isMounted) {
+          this.setState({ interiorElementRAF: interiorElement });
+        }
+      });
+    }
+  }
+
+  /**
+   * This is really ugly, but the tooltip needs to be outside of the normal
+   * DOM heirarchy so it isn't clipped by some arbitrary stacking context.
+   */
```

Comment:
I tried it locally, and I was still running into stacking issues. `fixed` would break out of the overflow, but would still obey some stacking context rules.